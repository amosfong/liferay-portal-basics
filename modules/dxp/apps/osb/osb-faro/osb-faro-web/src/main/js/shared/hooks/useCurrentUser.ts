import {fetchCurrentUser} from 'shared/actions/users';
import {useDispatch, useSelector} from 'react-redux';
import {useEffect} from 'react';
import {User} from 'shared/util/records';

/**
 * Used only on the first time on App.tsx to fetch data from backend.
 * To get currentUser, you can use WithCurrentUser (for HOC) or useCurrentUser (for HOOK).
 */
export const useFetchCurrentUser = (initialGroupId: string = '0') => {
	const currentUser = useSelector<any, any>(state =>
		state.get('currentUser')
	);
	const error = currentUser.get('error');
	const loading = currentUser.get('loading');

	const data = useCurrentUser();

	const dispatch = useDispatch();

	useEffect(() => {
		let groupId = '0';

		if (initialGroupId && initialGroupId !== 'add') {
			groupId = initialGroupId;
		}

		dispatch(fetchCurrentUser(groupId));
	}, [initialGroupId]);

	return {
		data,
		error,
		loading
	};
};

/**
 * Get currentUser from redux store.
 */
export const useCurrentUser = (): User => {
	const currentUserId = useSelector<any, any>(state =>
		state.getIn(['currentUser', 'data'])
	);
	const data: User = useSelector<any, any>(state =>
		state.getIn(['users', currentUserId, 'data'])
	);

	const newUser = new User({
		emailAddress: '',
		id: '',
		name: '',
		roleName: '',
		status: 1
	});

	return data || newUser;
};
