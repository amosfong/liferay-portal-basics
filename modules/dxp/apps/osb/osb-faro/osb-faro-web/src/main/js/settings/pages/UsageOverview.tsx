import BasePage from 'settings/components/BasePage';
import Card from 'shared/components/Card';
import ClayAlert from '@clayui/alert';
import ClayLayout from '@clayui/layout';
import moment from 'moment';
import React, {useState} from 'react';
import {AlertTypes} from 'shared/components/Alert';
import {compose, withProject} from 'shared/hoc';
import {CUSTOM_DATE_FORMAT, formatDateToTimeZone} from 'shared/util/date';
import {
	formatPlanData,
	isBasicPlan,
	PLAN_TYPES,
	PLANS
} from 'shared/util/subscriptions';
import {KnownIndividualsSession} from 'settings/components/usage-overview/KnownIndividualsSession';
import {PageViewsSession} from 'settings/components/usage-overview/PageViewsSession';
import {sub} from 'shared/util/lang';
import {SubscriptionDetails} from 'settings/components/usage-overview/SubscriptionDetails';
import {SubscriptionStatuses} from 'shared/util/constants';
import {Text} from '@clayui/core';
import {useCurrentUser} from 'shared/hooks/useCurrentUser';
import {useTimeZone} from 'shared/hooks/useTimeZone';

const subscriptionStatuses = admin => ({
	[SubscriptionStatuses.Approaching]: {
		message: admin
			? Liferay.Language.get(
					'usage-limit-is-approaching.-please-contact-your-sales-representative-at-the-earliest-convenience'
			  )
			: Liferay.Language.get(
					'usage-limit-is-approaching.-please-contact-your-workspace-administrator-at-the-earliest-convenience'
			  ),
		title: Liferay.Language.get('alert')
	},
	[SubscriptionStatuses.Over]: {
		message: admin
			? Liferay.Language.get(
					'usage-limit-exceeded.-please-contact-your-sales-representative-to-upgrade-the-plan'
			  )
			: Liferay.Language.get(
					'usage-limit-exceeded.-please-contact-your-workspace-administrator-to-upgrade-the-plan'
			  ),
		title: Liferay.Language.get('alert')
	}
});

const getAlertStatusCode = currentPlan => {
	const individuals = currentPlan.metrics.get('individuals');
	const pageViews = currentPlan.metrics.get('pageViews');

	if (individuals.status !== pageViews.status) {
		if (individuals.status !== SubscriptionStatuses.Ok) {
			return individuals.status;
		}

		if (
			individuals.status === SubscriptionStatuses.Ok ||
			pageViews.status === SubscriptionStatuses.Over
		) {
			return pageViews.status;
		}
	}

	return null;
};

export const UsageOverview = ({groupId, project}) => {
	const [showAlert, setShowAlert] = useState(true);
	const currentUser = useCurrentUser();

	const admin = currentUser.isAdmin();
	const currentPlan = formatPlanData(project.faroSubscription);
	const {timeZoneId} = useTimeZone();
	const planType =
		PLAN_TYPES[currentPlan.name] || PLAN_TYPES[PLANS.basic.name];

	let pageActions = [];

	if (currentUser.isAdmin()) {
		pageActions = [
			{
				displayType: 'primary',
				href: 'https://support.liferay.com/',
				icon: {
					symbol: 'shortcut'
				},
				label: Liferay.Language.get('manage-subscriptions'),
				target: '_blank'
			}
		];
	}

	const alertContent = subscriptionStatuses(admin)[
		getAlertStatusCode(currentPlan)
	];

	return (
		<BasePage
			groupId={groupId}
			key='UsageOverview'
			pageActions={pageActions}
			pageDescription={Liferay.Language.get(
				'plans-are-limited-by-the-total-amount-of-individuals-and-page-views'
			)}
			pageTitle={Liferay.Language.get('subscription-&-usage')}
		>
			{showAlert && alertContent && (
				<ClayLayout.Row>
					<ClayLayout.Col xl={12}>
						<ClayAlert
							displayType={AlertTypes.Warning}
							onClose={() => setShowAlert(false)}
							title={alertContent.title}
						>
							{alertContent.message}
						</ClayAlert>
					</ClayLayout.Col>
				</ClayLayout.Row>
			)}

			<ClayLayout.Row>
				<ClayLayout.Col xl={8}>
					<Card>
						<Card.Header>
							<Card.Title>
								{Liferay.Language.get('usage-limits')}
							</Card.Title>
						</Card.Header>

						<Card.Body>
							{isBasicPlan(currentPlan) ? (
								<p className='mb-0'>
									<Text color='secondary' size={3}>
										{Liferay.Language.get(
											'when-either-limit-is-exceeded-the-current-plan-will-have-to-be-upgraded-to-business-or-enterprise'
										)}
									</Text>
								</p>
							) : (
								<>
									<p>
										<Text color='secondary' size={3}>
											{Liferay.Language.get(
												'when-either-limit-is-exceeded-the-current-plan-will-either-have-to-be-upgraded-or-add-ons-will-have-to-be-purchased-to-accommodate-the-overage'
											)}
										</Text>
									</p>

									<p
										className='mb-0'
										data-testid='next-anniversary-date'
									>
										<Text color='secondary' size={3}>
											{sub(
												Liferay.Language.get(
													'plan-usage-resets-on-x'
												),
												[
													<b key='DATE'>
														{formatDateToTimeZone(
															moment(
																currentPlan.startDate
															).add(1, 'year'),
															CUSTOM_DATE_FORMAT,
															timeZoneId
														)}
													</b>
												],
												false
											)}
										</Text>
									</p>
								</>
							)}

							<KnownIndividualsSession
								currentPlan={currentPlan}
							/>

							<PageViewsSession currentPlan={currentPlan} />
						</Card.Body>
					</Card>
				</ClayLayout.Col>

				<ClayLayout.Col xl={4}>
					<SubscriptionDetails
						currentPlan={currentPlan}
						planType={planType}
					/>
				</ClayLayout.Col>
			</ClayLayout.Row>
		</BasePage>
	);
};

export default compose(withProject)(UsageOverview);
