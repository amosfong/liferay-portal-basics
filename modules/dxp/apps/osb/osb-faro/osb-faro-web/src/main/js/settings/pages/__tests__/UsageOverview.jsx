import * as data from 'test/data';
import mockStore from 'test/mock-store';
import React from 'react';
import {cleanup, getByTestId, render} from '@testing-library/react';
import {fromJS} from 'immutable';
import {Project} from 'shared/util/records';
import {Provider} from 'react-redux';
import {StaticRouter} from 'react-router';
import {SubscriptionStatuses, UserRoleNames} from 'shared/util/constants';
import {UsageOverview} from '../UsageOverview';
import {useCurrentUser} from 'shared/hooks/useCurrentUser';
import {User} from 'shared/util/records';

jest.unmock('react-dom');

jest.mock('shared/hooks/useTimeZone', () => ({
	useTimeZone: () => ({timeZoneId: 'UTC'})
}));

jest.mock('shared/hooks/useCurrentUser', () => ({
	useCurrentUser: jest.fn()
}));

const defaultProps = {
	groupId: '23',
	project: new Project(
		data.mockProject(23, {
			faroSubscription: fromJS(data.mockSubscription())
		})
	)
};

const WrappedComponent = props => (
	<Provider store={mockStore()}>
		<StaticRouter>
			<UsageOverview {...props} />
		</StaticRouter>
	</Provider>
);

describe('UsageOverview', () => {
	it('should render', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const {container} = render(<WrappedComponent {...defaultProps} />);

		expect(container).toMatchSnapshot();
	});

	it('should render with a warning type and danger type warning if one metric is approaching limit and the other is over', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						individualsStatus: SubscriptionStatuses.Approaching,
						pageViewsStatus: SubscriptionStatuses.Over
					})
				)
			})
		);

		const {container} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(container.querySelector('.alert-warning')).toBeInTheDocument();
	});

	it('should render with an approaching limit warning if a metric is approaching plan limit', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						pageViewsStatus: SubscriptionStatuses.Approaching
					})
				)
			})
		);

		const {container} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(container.querySelector('.alert-warning')).toBeInTheDocument();
	});

	it('should render the total number of INDIVIDUALS since last anniversary and the percentage used in the plan', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						individualsCountSinceLastAnniversary: 1000,
						individualsStatus: SubscriptionStatuses.Ok
					})
				)
			})
		);

		const {getByText} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(getByText('KNOWN INDIVIDUALS')).toBeInTheDocument();

		expect(
			getByText(
				'Active users logged on your DXP instance have been tracked by Analytics Cloud since Jul 08, 2018.'
			)
		).toBeInTheDocument();

		expect(
			getByText('104,000 known individuals are available.')
		).toBeInTheDocument();

		expect(
			getByText('1,000 of 105,000 - 1% known individual was used.')
		).toBeInTheDocument();
	});

	it('should display the plan count percentage as 100% for INDIVIDUALS when count is higher than limit', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						individualsCountSinceLastAnniversary: 115000
					})
				)
			})
		);

		const {getByText} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(
			getByText('115,000 of 105,000 - 100% known individuals were used.')
		).toBeInTheDocument();
	});

	it('should display the plan count percentage as 100% for PAGE VIEWS when count is higher than limit', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						pageViewsCountSinceLastAnniversary: 8000000
					})
				)
			})
		);

		const {getByText} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(
			getByText('8,000,000 of 7,000,000 - 100% page views were used.')
		).toBeInTheDocument();
	});

	it('should display the limit of INDIVIDUALS and PAGE VIEWS. Also, it should render a warning if INDIVIDUALS is over the limit. Also, it should render the current plan name.', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						individualsStatus: SubscriptionStatuses.Over
					})
				)
			})
		);

		const {container, getByText} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(container.querySelector('.alert-warning')).toBeInTheDocument();

		expect(getByText('Enterprise')).toBeInTheDocument();
	});

	it('should render the total of PAGE VIEWS since the last anniversary and the percentage used in the plan', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						individualsStatus: SubscriptionStatuses.Ok,
						pageViewsCountSinceLastAnniversary: 111123
					})
				)
			})
		);

		const {getByText} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(getByText('PAGE VIEWS')).toBeInTheDocument();

		expect(
			getByText(
				'Active users logged on your DXP instance have been tracked by Analytics Cloud since Jul 08, 2018.'
			)
		).toBeInTheDocument();

		expect(
			getByText('111,123 of 7,000,000 - 1.6% page views were used.')
		).toBeInTheDocument();

		expect(
			getByText('6,888,877 page views are available.')
		).toBeInTheDocument();
	});

	it('should render with an overage warning if the PAGE VIEWS metric has exceeded the plan limit', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						pageViewsStatus: SubscriptionStatuses.Over
					})
				)
			})
		);

		const {container} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(container.querySelector('.alert-warning')).toBeInTheDocument();
	});

	it('should render with a member-specific message overage warning if a metric is approaching plan limit and the user is a member role', () => {
		useCurrentUser.mockImplementation(() => ({
			isAdmin: () => false
		}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						pageViewsStatus: SubscriptionStatuses.Approaching
					})
				)
			})
		);

		const {container, getByText} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(container.querySelector('.alert-warning')).toBeInTheDocument();
		expect(
			getByText(
				'Usage limit is approaching. Please contact your workspace administrator at the earliest convenience.'
			)
		).toBeInTheDocument();
	});

	it('not renders next anniversary date for BASIC PLAN', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						name: 'Liferay Analytics Cloud Basic'
					})
				)
			})
		);

		const {queryByTestId} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(queryByTestId('next-anniversary-date')).toBeNull();
	});

	it('renders next anniversary date for ENTERPRISE PLAN', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						name: 'Liferay Analytics Cloud Enterprise'
					})
				)
			})
		);

		const {queryByTestId} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(queryByTestId('next-anniversary-date').textContent).toEqual(
			'Plan usage resets on Jul 08, 2019.'
		);
	});

	it('renders next anniversary date for BUSINESS PLAN', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						name: 'Liferay Analytics Cloud Business'
					})
				)
			})
		);

		const {queryByTestId} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(queryByTestId('next-anniversary-date').textContent).toEqual(
			'Plan usage resets on Jul 08, 2019.'
		);
	});

	it('not renders management button for Member view', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(data.mockSubscription())
			})
		);

		const {queryByText} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(
			queryByText('Manage Subscriptions(Opens a new window)')
		).toBeNull();
	});

	it('not renders management button for Admin view', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(data.mockSubscription())
			})
		);

		const {queryByText} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(queryByText('Manage Subscriptions')).toBeInTheDocument();
	});

	it('renders current usage message for BASIC PLAN', () => {
		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						name: 'Liferay Analytics Cloud Basic'
					})
				)
			})
		);

		const {queryByText} = render(
			<WrappedComponent
				{...defaultProps}
				currentUser={
					new User(data.mockUser(0, {roleName: UserRoleNames.Member}))
				}
				project={mockProject}
			/>
		);

		expect(
			queryByText(
				'When either limit is exceeded, the current plan will have to be upgraded to Business or Enterprise.'
			)
		).toBeInTheDocument();
	});

	it('renders current usage message for BUSINESS PLAN', () => {
		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						name: 'Liferay Analytics Cloud Business'
					})
				)
			})
		);

		const {queryByText} = render(
			<WrappedComponent
				{...defaultProps}
				currentUser={
					new User(data.mockUser(0, {roleName: UserRoleNames.Member}))
				}
				project={mockProject}
			/>
		);

		expect(
			queryByText(
				'When either limit is exceeded, the current plan will either have to be upgraded or add-ons will have to be purchased to accommodate the overage.'
			)
		).toBeInTheDocument();
	});

	it('renders current usage message for ENTERPRISE PLAN', () => {
		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						name: 'Liferay Analytics Cloud Enterprise'
					})
				)
			})
		);

		const {queryByText} = render(
			<WrappedComponent
				{...defaultProps}
				currentUser={
					new User(data.mockUser(0, {roleName: UserRoleNames.Member}))
				}
				project={mockProject}
			/>
		);

		expect(
			queryByText(
				'When either limit is exceeded, the current plan will either have to be upgraded or add-ons will have to be purchased to accommodate the overage.'
			)
		).toBeInTheDocument();
	});
});

describe('Subscription Details', () => {
	afterEach(cleanup);

	it('renders subscription details for BASIC PLAN', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						name: 'Liferay Analytics Cloud Basic'
					})
				)
			})
		);

		const {container, queryByText} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(queryByText('PURCHASED ADD-ONS')).toBeNull();

		expect(
			getByTestId(container, 'subscription-details')
		).toMatchSnapshot();
	});

	it('renders subscription details for BUSINESS PLAN', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						name: 'Liferay Analytics Cloud Business'
					})
				)
			})
		);

		const {container, queryByText} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(queryByText('PURCHASED ADD-ONS')).toBeInTheDocument();

		expect(
			getByTestId(container, 'subscription-details')
		).toMatchSnapshot();
	});

	it('renders subscription details for ENTERPRISE PLAN', () => {
		useCurrentUser.mockImplementation(() => ({isAdmin: () => true}));

		const mockProject = new Project(
			data.mockProject(23, {
				faroSubscription: fromJS(
					data.mockSubscription({
						name: 'Liferay Analytics Cloud Enterprise'
					})
				)
			})
		);

		const {container, queryByText} = render(
			<WrappedComponent {...defaultProps} project={mockProject} />
		);

		expect(queryByText('PURCHASED ADD-ONS')).toBeInTheDocument();

		expect(
			getByTestId(container, 'subscription-details')
		).toMatchSnapshot();
	});
});
