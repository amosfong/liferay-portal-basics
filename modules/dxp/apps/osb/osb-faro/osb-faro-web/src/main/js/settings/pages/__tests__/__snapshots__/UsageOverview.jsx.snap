// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription Details renders subscription details for BASIC PLAN 1`] = `
<div
  class="card card-root"
  data-testid="subscription-details"
>
  <div
    class="card-header py-2"
  >
    <div
      class="card-title"
    >
      <span
        class="text-3 text-secondary"
      >
        SUBSCRIPTION DETAILS
      </span>
    </div>
  </div>
  <hr
    class="m-0"
  />
  <div
    class="card-body"
  >
    <div
      class="mb-2"
    >
      <span
        class="text-3 font-weight-semi-bold"
      >
        Basic
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Individuals
      </span>
      <span
        class="text-3 text-secondary"
      >
        1,000
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Page Views
      </span>
      <span
        class="text-3 text-secondary"
      >
        300,000
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Workspace Users
      </span>
      <span
        class="text-3 text-secondary"
      >
        Unlimited
      </span>
    </div>
  </div>
</div>
`;

exports[`Subscription Details renders subscription details for BUSINESS PLAN 1`] = `
<div
  class="card card-root"
  data-testid="subscription-details"
>
  <div
    class="card-header py-2"
  >
    <div
      class="card-title"
    >
      <span
        class="text-3 text-secondary"
      >
        SUBSCRIPTION DETAILS
      </span>
    </div>
  </div>
  <hr
    class="m-0"
  />
  <div
    class="card-body pb-0"
  >
    <div
      class="mb-2"
    >
      <span
        class="text-3 font-weight-semi-bold"
      >
        Business
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Individuals
      </span>
      <span
        class="text-3 text-secondary"
      >
        10,000
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Page Views
      </span>
      <span
        class="text-3 text-secondary"
      >
        5,000,000
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Workspace Users
      </span>
      <span
        class="text-3 text-secondary"
      >
        Unlimited
      </span>
    </div>
  </div>
  <div
    class="card-header pb-2"
  >
    <div
      class="card-title"
    >
      <span
        class="text-3 text-secondary"
      >
        PURCHASED ADD-ONS
      </span>
    </div>
  </div>
  <div
    class="card-body"
  >
    <hr
      class="mt-0"
    />
    <div
      class="mb-2"
    >
      <span
        class="text-3 text-secondary"
      >
        Tailor limits to business needs. Incrementally increase individual or page view limits as needed without committing to a new plan.
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Individuals
      </span>
      <span
        class="text-3 text-secondary"
      >
        10,000
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Page Views
      </span>
      <span
        class="text-3 text-secondary"
      >
        5,000,000
      </span>
    </div>
  </div>
</div>
`;

exports[`Subscription Details renders subscription details for ENTERPRISE PLAN 1`] = `
<div
  class="card card-root"
  data-testid="subscription-details"
>
  <div
    class="card-header py-2"
  >
    <div
      class="card-title"
    >
      <span
        class="text-3 text-secondary"
      >
        SUBSCRIPTION DETAILS
      </span>
    </div>
  </div>
  <hr
    class="m-0"
  />
  <div
    class="card-body pb-0"
  >
    <div
      class="mb-2"
    >
      <span
        class="text-3 font-weight-semi-bold"
      >
        Enterprise
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Individuals
      </span>
      <span
        class="text-3 text-secondary"
      >
        100,000
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Page Views
      </span>
      <span
        class="text-3 text-secondary"
      >
        60,000,000
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Workspace Users
      </span>
      <span
        class="text-3 text-secondary"
      >
        Unlimited
      </span>
    </div>
  </div>
  <div
    class="card-header pb-2"
  >
    <div
      class="card-title"
    >
      <span
        class="text-3 text-secondary"
      >
        PURCHASED ADD-ONS
      </span>
    </div>
  </div>
  <div
    class="card-body"
  >
    <hr
      class="mt-0"
    />
    <div
      class="mb-2"
    >
      <span
        class="text-3 text-secondary"
      >
        Tailor limits to business needs. Incrementally increase individual or page view limits as needed without committing to a new plan.
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Individuals
      </span>
      <span
        class="text-3 text-secondary"
      >
        10,000
      </span>
    </div>
    <div
      class="d-flex justify-content-between mb-1"
    >
      <span
        class="text-3 text-secondary font-weight-semi-bold"
      >
        Page Views
      </span>
      <span
        class="text-3 text-secondary"
      >
        5,000,000
      </span>
    </div>
  </div>
</div>
`;

exports[`UsageOverview should render 1`] = `
<div>
  <div
    class="settings-base-page-root"
  >
    <div
      class="maintenance-alert-root"
    />
    <div
      class="content-header has-page-actions"
    >
      <div
        class="header-text"
      >
        <div
          class="d-flex"
        >
          <h3
            class="title-text text-truncate"
          >
            <span
              class="text-truncate"
              data-tooltip="false"
              title=""
            >
              Subscription & Usage
            </span>
          </h3>
        </div>
        <div
          class="description"
        >
          Plans are limited by the total amount of individuals and page views.
        </div>
      </div>
      <div
        class="page-actions-container"
      >
        <a
          class="button-root btn btn-primary"
          href="https://support.liferay.com/"
          rel="noreferrer noopener"
          target="_blank"
        >
          <svg
            class="lexicon-icon lexicon-icon-shortcut mr-2"
            role="presentation"
          >
            <use
              href="#shortcut"
            />
          </svg>
          Manage Subscriptions
          <span
            class="sr-only"
          >
            (Opens a new window)
          </span>
        </a>
      </div>
    </div>
    <div>
      <div
        class="row"
      >
        <div
          class="col-xl-8"
        >
          <div
            class="card card-root"
          >
            <div
              class="card-header"
            >
              <div
                class="card-title"
              >
                Usage Limits
              </div>
            </div>
            <div
              class="card-body"
            >
              <p>
                <span
                  class="text-3 text-secondary"
                >
                  When either limit is exceeded, the current plan will either have to be upgraded or add-ons will have to be purchased to accommodate the overage.
                </span>
              </p>
              <p
                class="mb-0"
                data-testid="next-anniversary-date"
              >
                <span
                  class="text-3 text-secondary"
                >
                  Plan usage resets on 
                  <b>
                    Jul 08, 2019
                  </b>
                  .
                </span>
              </p>
              <div
                class="mt-4 mb-5"
              >
                <span
                  class="text-3 text-secondary font-weight-semi-bold"
                >
                  KNOWN INDIVIDUALS
                </span>
                <hr
                  class="my-2"
                />
                <span
                  class="text-3 text-secondary"
                >
                  Active users logged on your DXP instance have been tracked by Analytics Cloud since Jul 08, 2018.
                </span>
                <div
                  class="mt-3"
                >
                  <div
                    class="d-flex justify-content-between mb-1"
                  >
                    <span
                      class="text-3 text-secondary"
                    >
                      CURRENT USAGE
                    </span>
                    <span
                      class="text-3 text-secondary"
                    >
                      2,057 of 105,000 - 2% known individuals were used.
                    </span>
                  </div>
                  <div
                    class="recharts-responsive-container"
                    style="width: 100%; height: 24px; min-width: 0;"
                  />
                  <span
                    class="text-3 text-secondary"
                  >
                    102,943 known individuals are available.
                  </span>
                  <div
                    class="mt-4"
                  >
                    <div
                      class="mb-1"
                    >
                      <span
                        class="text-3 text-secondary"
                      >
                        INDIVIDUALS BREAKDOWN
                      </span>
                    </div>
                    <div
                      class="recharts-responsive-container"
                      style="width: 100%; height: 24px; min-width: 0;"
                    />
                  </div>
                </div>
              </div>
              <span
                class="text-3 text-secondary font-weight-semi-bold"
              >
                PAGE VIEWS
              </span>
              <hr
                class="my-2"
              />
              <span
                class="text-3 text-secondary"
              >
                Total page views have been tracked by Analytics Cloud since Jul 08, 2018.
              </span>
              <div
                class="mt-3"
              >
                <div
                  class="d-flex justify-content-between mb-1"
                >
                  <span
                    class="text-3 text-secondary"
                  >
                    CURRENT USAGE
                  </span>
                  <span
                    class="text-3 text-secondary"
                  >
                    100,023 of 7,000,000 - 1.4% page views were used.
                  </span>
                </div>
                <div
                  class="recharts-responsive-container"
                  style="width: 100%; height: 24px; min-width: 0;"
                />
                <span
                  class="text-3 text-secondary"
                >
                  6,899,977 page views are available.
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-xl-4"
        >
          <div
            class="card card-root"
            data-testid="subscription-details"
          >
            <div
              class="card-header py-2"
            >
              <div
                class="card-title"
              >
                <span
                  class="text-3 text-secondary"
                >
                  SUBSCRIPTION DETAILS
                </span>
              </div>
            </div>
            <hr
              class="m-0"
            />
            <div
              class="card-body pb-0"
            >
              <div
                class="mb-2"
              >
                <span
                  class="text-3 font-weight-semi-bold"
                >
                  Enterprise
                </span>
              </div>
              <div
                class="d-flex justify-content-between mb-1"
              >
                <span
                  class="text-3 text-secondary font-weight-semi-bold"
                >
                  Individuals
                </span>
                <span
                  class="text-3 text-secondary"
                >
                  100,000
                </span>
              </div>
              <div
                class="d-flex justify-content-between mb-1"
              >
                <span
                  class="text-3 text-secondary font-weight-semi-bold"
                >
                  Page Views
                </span>
                <span
                  class="text-3 text-secondary"
                >
                  60,000,000
                </span>
              </div>
              <div
                class="d-flex justify-content-between mb-1"
              >
                <span
                  class="text-3 text-secondary font-weight-semi-bold"
                >
                  Workspace Users
                </span>
                <span
                  class="text-3 text-secondary"
                >
                  Unlimited
                </span>
              </div>
            </div>
            <div
              class="card-header pb-2"
            >
              <div
                class="card-title"
              >
                <span
                  class="text-3 text-secondary"
                >
                  PURCHASED ADD-ONS
                </span>
              </div>
            </div>
            <div
              class="card-body"
            >
              <hr
                class="mt-0"
              />
              <div
                class="mb-2"
              >
                <span
                  class="text-3 text-secondary"
                >
                  Tailor limits to business needs. Incrementally increase individual or page view limits as needed without committing to a new plan.
                </span>
              </div>
              <div
                class="d-flex justify-content-between mb-1"
              >
                <span
                  class="text-3 text-secondary font-weight-semi-bold"
                >
                  Individuals
                </span>
                <span
                  class="text-3 text-secondary"
                >
                  10,000
                </span>
              </div>
              <div
                class="d-flex justify-content-between mb-1"
              >
                <span
                  class="text-3 text-secondary font-weight-semi-bold"
                >
                  Page Views
                </span>
                <span
                  class="text-3 text-secondary"
                >
                  5,000,000
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
